### PR TITLE
Fix: Complete rewrite

### LINKED ISSUE #10: Cookie accept modal on consent subdomain not fully styled
## Issue
A modal is displayed when visiting YouTube for the first time from a location in the EU to explain and select cookie preferences.

If the first page visited is a video or the YouTube homepage, this is displayed as a page overlay modal and is styled correctly. If however, the first page hit is a channel or hashtag, the user is instead redirected to an acceptance page - `https://consent.youtube.com/m?continue=...`

For the former use cases, the styling is applied correctly, but for the consent subdomain, text and button styles (including sign-in) are not applied.

### Actual
First hit type - Hashtag: [`/hashtag/catppuccin` redirection](https://consent.youtube.com/m?continue=https%3A%2F%2Fwww.youtube.com%2Fhashtag%2Fcatppuccin%3Fcbrd%3D1&gl=GB&m=0&pc=yt&hl=en-GB&src=1)
image

First hit type - Channel: [`/channel/TheLinuxCast` redirection](https://consent.youtube.com/m?continue=https%3A%2F%2Fwww.youtube.com%2Fc%2FTheLinuxCast%3Fcbrd%3D1&gl=GB&m=0&pc=yt&hl=en-GB&src=1)
image

### Expected
First hit type - Video
image

First hit type - Homepage
image

### LINKED ISSUE #11: Error pages partially misstyled
## Issue
When a user hits an error page, either YouTube styled, or Google styled, various elements are misstyled, or have image aliasing issues.

- YouTube type not found error pages have black text with white drop shadow
- YouTube type internal error pages are missing most styles
- Google type error pages have black text and aliasing issues on the broken robot PNG

### Consent subdomain [error page](https://consent.youtube.com/m) (Google styled)
image

### Channel [error page](https://www.youtube.com/c/totally-not-a-real-channel) (YouTube not found styled)
image

### Internal [error page](https://www.youtube.com/live_chat) (YouTube internal error styled)
image

### LINKED ISSUE #12: Text type with pages incorrectly styled text
## Issue
When accessing a text-based page on YouTube, the body background colour is applied, but the text is not styled.

### [Robots](https://www.youtube.com/robots.txt) - Plaintext Type
image

### [Sitemap](https://www.youtube.com/product/sitemap.xml) - XML Type
image

### LINKED ISSUE #18: Icons color in playback setting menu might be wrong
I believe that the icon should have same color with the text label because it has barely seen.

Screenshot 2022-11-08 at 13 13 31

### LINKED ISSUE #20: Catppuccin Latte (light theme) on YouTube broke the contrast on subscribe and other buttons.
My YouTube setting is on light theme. I have been using Catppuccin Latte for months with no issue, but today the style got updated, and seems to have broken the contrast on buttons below the video player, including the subscribe button.

YouTube with no extra CSS, for comparison:
image

YouTube with Catppuccin Latte:
image

YouTube with a band-aid code fix that I wrote, that is what I would expect it to look like:
image

The code I used in an extra style to band-aid patch the issue:
```css
.yt-spec-button-shape-next--mono.yt-spec-button-shape-next--filled,
.yt-spec-button-shape-next--mono.yt-spec-button-shape-next--tonal,
tp-yt-paper-button.ytd-subscribe-button-renderer {
    color: var(--yt-spec-text-primary)!important;
}
```

I'm unsure which color(s) would be ideal for usage out of the Catppuccin palette, or if this is pulling the correct color with that variable (I haven't had a moment to dig into the code myself to see what's what) but hopefully the selectors (which I just pulled out of the inspector) should help make it easy to fix. I think it's actually just the second one that matters in this case -- again, I just did a quick band-aid fix and grabbed the entire selector from the inspector.

Let me know if any further context is needed, I'd be happy to provide any information I can. Thanks.

### LINKED ISSUE #25: Fix the info card that commonly appears on news videos.
```
ytd-info-panel-content-renderer[background="BLUE"] .content.ytd-info-panel-content-renderer {
    background: var(--second-background) !important;
    }
```
This code block makes it so that you can read the text.

### LINKED ISSUE #27: Latest version will not load.
https://github.com/catppuccin/youtube/commit/2f551152678304d260e30aab0dcbb907aa0be334 is a breaking change, seemingly due to 2 extra spaces at the start of a few dozen lines

https://github.com/catppuccin/youtube/blob/009b7a1778ebbbd365d15e2cf9b2b9075e9705f1/src/YouTubeCatppuccinMocha.user.css#L330
https://github.com/catppuccin/youtube/blob/009b7a1778ebbbd365d15e2cf9b2b9075e9705f1/src/YouTubeCatppuccinMocha.user.css#L872
https://github.com/catppuccin/youtube/blob/009b7a1778ebbbd365d15e2cf9b2b9075e9705f1/src/YouTubeCatppuccinMocha.user.css#L1094

edit: Seems to have caused indenting problems all over.

### LINKED ISSUE #29: Style leaking to other websites
It appears as if the style is leaking onto other websites as well.
Could be a RegEx Issue.

### LINKED ISSUE #32: inccorrect logo color
### Bug:
logo of youtube in top left is incorrect:
image
however if i go to some video and open side bar logo looks correct:
image

### Additional Information
tested on mocha style

### LINKED ISSUE #33: VR.youtube only partially styled
The promotional page for Youtube VR only partially covers styling components,often with text left illegible.

Examples:
Screenshot 2023-03-02 at 12 50 35
Screenshot 2023-03-02 at 12 50 40
Screenshot 2023-03-02 at 12 50 53
Screenshot 2023-03-02 at 12 50 57

URL: [Youtube VIR](https://vr.youtube.com/)

### PR DESCRIPTION
**Pull request by isabelroses**
This is a complete YouTube rewrite for the themes and documentation. 

I am willing to help maintain it too if it is necessary.

### CODE REVIEW

File: src/YouTubeCatppuccinFrappe.user.css [hunk 1]
```diff
@@ -1,11 +1,11 @@
 /* ==UserStyle==
 @name YouTube Catppuccin-Frappe
-@namespace github.com/catppuccin/YouTube
-@homepageURL https://github.com/catppuccin/YouTube
-@version 1.3.7
-@updateURL https://github.com/catppuccin/YouTube/raw/main/src/YouTubeCatppuccinFrappe.user.css
+@namespace github.com/isabelroses/YouTube
```
Comment (nullishamy) [inline src/YouTubeCatppuccinFrappe.user.css]: This should still link to ctp/youtube.

File: src/YouTubeCatppuccinFrappe.user.css [hunk 2]
```diff
@@ -4187,36 +3665,20 @@
...
         height: 18px !important;
         padding-left: 18px !important;
     }
-
     .sponsorBlockTooltip {
         background-color: var(--hover-background) !important;
     }
-
-    /*Fix like, dislike and share icons and text using wrong color*/
-    .yt-spec-button-shape-next__icon,
```
Comment (nullishamy) [inline src/YouTubeCatppuccinFrappe.user.css:4209]: Why was this fix removed?

File: src/YouTubeCatppuccinLatte.user.css [hunk 3]
```diff
@@ -1,11 +1,11 @@
 /* ==UserStyle==
 @name YouTube Catppuccin-Latte
-@namespace github.com/catppuccin/YouTube
-@homepageURL https://github.com/catppuccin/YouTube
-@version 1.3.7
-@updateURL https://github.com/catppuccin/YouTube/raw/main/src/YouTubeCatppuccinLatte.user.css
+@namespace github.com/isabelroses/YouTube
```
Comment (nullishamy) [inline src/YouTubeCatppuccinLatte.user.css]: Same as above, should be ctp/youtube.

File: src/YouTubeCatppuccinMacchiato.user.css [hunk 4]
```diff
@@ -1,11 +1,11 @@
 /* ==UserStyle==
 @name YouTube Catppuccin-Macchiato
-@namespace github.com/catppuccin/YouTube
-@homepageURL https://github.com/catppuccin/YouTube
-@version 1.3.7
-@updateURL https://github.com/catppuccin/YouTube/raw/main/src/YouTubeCatppuccinMacchiato.user.css
+@namespace github.com/isabelroses/YouTube
```
Comment (nullishamy) [inline src/YouTubeCatppuccinMacchiato.user.css]: Same as above, should be ctp/youtube.

File: src/YouTubeCatppuccinMocha.user.css [hunk 5]
```diff
@@ -1,11 +1,11 @@
 /* ==UserStyle==
 @name YouTube Catppuccin-Mocha
-@namespace github.com/catppuccin/YouTube
-@homepageURL https://github.com/catppuccin/YouTube
-@version 1.3.7
-@updateURL https://github.com/catppuccin/YouTube/raw/main/src/YouTubeCatppuccinMocha.user.css
+@namespace github.com/isabelroses/YouTube
```
Comment (nullishamy) [inline src/YouTubeCatppuccinMocha.user.css]: Same as above, should be ctp/youtube.

File: src/YouTubeCatppuccinLatte.user.css [hunk 6]
```diff
@@ -4187,36 +3662,20 @@
...
         height: 18px !important;
         padding-left: 18px !important;
     }
-
     .sponsorBlockTooltip {
         background-color: var(--hover-background) !important;
     }
-
-    /*Fix like, dislike and share icons and text using wrong color*/
-    .yt-spec-button-shape-next__icon,
```
Comment (nullishamy) [inline src/YouTubeCatppuccinLatte.user.css:4209]: Same as above, why was this fix removed?

File: src/YouTubeCatppuccinMacchiato.user.css [hunk 7]
```diff
@@ -4187,36 +3661,20 @@
...
         height: 18px !important;
         padding-left: 18px !important;
     }
-
     .sponsorBlockTooltip {
         background-color: var(--hover-background) !important;
     }
-
-    /*Fix like, dislike and share icons and text using wrong color*/
-    .yt-spec-button-shape-next__icon,
```
Comment (nullishamy) [inline src/YouTubeCatppuccinMacchiato.user.css:4209]: Same as above, why was this fix removed?

File: src/YouTubeCatppuccinMocha.user.css [hunk 8]
```diff
@@ -4187,36 +3661,20 @@
...
         height: 18px !important;
         padding-left: 18px !important;
     }
-
     .sponsorBlockTooltip {
         background-color: var(--hover-background) !important;
     }
-
-    /*Fix like, dislike and share icons and text using wrong color*/
-    .yt-spec-button-shape-next__icon,
```
Comment (nullishamy) [inline src/YouTubeCatppuccinMocha.user.css:4209]: Same as above, why was this fix removed?

**Review by nullishamy — CHANGES_REQUESTED** [review]

File: src/YouTubeCatppuccinFrappe.user.css (hunk 1 above)
Comment (isabelroses) [reply src/YouTubeCatppuccinFrappe.user.css]: I shall fix that right now

File: src/YouTubeCatppuccinFrappe.user.css (hunk 2 above)
Comment (isabelroses) [reply src/YouTubeCatppuccinFrappe.user.css:4209]: this was a temporary fix and it now appears correctly.

**Comment by isabelroses** [comment]
So I put all the ctp links back within the code. And for this section ```/*Fix like, dislike and share icons and text using wrong color*/
    .yt-spec-button-shape-next__icon, ```
I removed it as it was a temporary fix and will work in the newer version. The following example is from the mocha port showing it does work:
image

File: src/YouTubeCatppuccinMocha.user.css (hunk 8 above)
Comment (isabelroses) [reply src/YouTubeCatppuccinMocha.user.css:4209]: this was a temporary fix and it now appears correctly.
image
example form mocha

File: src/YouTubeCatppuccinMacchiato.user.css (hunk 7 above)
Comment (isabelroses) [reply src/YouTubeCatppuccinMacchiato.user.css:4209]: this was a temporary fix and it now appears correctly.
image
example form mocha

File: src/YouTubeCatppuccinLatte.user.css (hunk 6 above)
Comment (isabelroses) [reply src/YouTubeCatppuccinLatte.user.css:4209]: this was a temporary fix and it now appears correctly.
image
example form mocha

**Comment by elkrien** [comment]
> I am willing to help maintain it too if it is necessary.

Would be great because I have no time to be involved anymore. We will need to wait for organizations admin to add You as a maintainer (I have no rights to do it).
@nullishamy @nekowinston - maybe You can help?

**Comment by nekowinston** [comment]
> > I am willing to help maintain it too if it is necessary.
> 
> Would be great because I have no time to be involved anymore. We will need to wait for organizations admin to add You as a maintainer (I have no rights to do it).

I've added @isabelroses as a maintainer. Thanks for letting us know @elkrien!

**Comment by isabelroses** [comment]
PR's that this fixes:
[fix: style leaking to other sites](https://github.com/catppuccin/youtube/pull/30) - caused by brackets, now fixed
[fix: Notification icon text color](https://github.com/catppuccin/youtube/pull/28) - tests suggest this is fixed
[fix: Improve comment readability](https://github.com/catppuccin/youtube/pull/23) - This is an older PR that I reverted some of the code back to
[fix: subscribe button](https://github.com/catppuccin/youtube/pull/36) and [fix: subscribe button](https://github.com/catppuccin/youtube/pull/35) - both my own PRs however this was something completely rewritten in the change and significant testing into the matter with help from [lighttigerXIV](https://github.com/lighttigerXIV)
[chore: update buttons shape](https://github.com/catppuccin/youtube/pull/26) - another change for text color of subscribe and join button for readability fixed with this change again
[fix: some buttons and player tooltips](https://github.com/catppuccin/youtube/pull/22) - buttons were fixed during rewrite and I just committed the change for the tooltips.
[fix: metadata card](https://github.com/catppuccin/youtube/pull/3) - fixed in rewrite
[fix: selection readablity](https://github.com/catppuccin/youtube/pull/1) - newer iterations of the code override this, reverting back to this

Issues that remain:
[error pages](https://github.com/catppuccin/youtube/issues/11) - partially fixed

**Comment by nullishamy** [comment]
> PR's that this fixes:

You can write `Closes <pr>` to have GitHub automatically close the issues when this is merged btw.

**Review by nullishamy — APPROVED** [review]

**Pull request merged**